test/functional/zmq_test.py

```diff
@@ -13,9 +13,33 @@
                                  hash256,
                                 )
 
+class ZMQSubscriber:
+    def __init__(self, socket, address):
+        self.address = address
+        self.socket = socket
+        self.sequence = 0
+
 class ZMQTest (BitcoinTestFramework):
     def set_test_params(self):
+        self.address = "tcp://127.0.0.1:28332"
+        self.num_blocks = 5
         self.num_nodes = 2
+        self.subscribers = {}
```

Comment:
I don't like adding these assignments to `set_test_params`.

- `num_blocks` and `subscribers` can be local variables in `run_test()`
- `address` should be a local variable in `setup_nodes()` and passed into the `ZMQSubscriber` constructor. Conceptually, `address` is the address of the ZMQ subscriber, not the test framework.

`set_test_params()` should be used for overriding the number of nodes, network topology, etc.

Reply:
Ok.